src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -400,7 +408,7 @@ int32_t SystemNative_GetDomainName(uint8_t* name, int32_t nameLength)
 #endif
 
     return getdomainname((char*)name, namelen);
-#elif HAVE_UNAME
+#elif HAVE_UNAME && !defined (__HAIKU__)
```

Comment:
Why isn't HAVE_UNAME suficient?

Reply:
Haiku lacks the domainname structure member.

Reply:
Can we modify the HAVE_UNAME check them to also look for that? Would anything break with any other uses? I'd really like to minimize the amount of code that references a specific OS and instead make it work as much as possible based on the capabilities of the platform.

Reply:
I know autotools can check for certain structure members, I'll check if CMake can.